core/src/ast/query.rs

```diff
@@ -58,6 +59,24 @@ pub enum TableFactor {
         /// Query planner result
         index: Option<IndexItem>,
     },
+    Derived {
+        subquery: Query,
+        alias: TableAlias,
+    },
+}
+impl TableFactor {
+    pub(crate) fn get_name(&self) -> Result<&String> {
+        match self {
+            Self::Table { .. } => Ok(Table::new(self)?.get_name()),
+            Self::Derived { alias, .. } => Ok(&alias.name),
+        }
+    }
+    pub(crate) fn get_alias(&self) -> Result<&String> {
+        match self {
+            Self::Table { .. } => Ok(Table::new(self)?.get_alias()),
+            Self::Derived { alias, .. } => Ok(&alias.name),
+        }
+    }
```

Comment:
Here is in the `ast` module, and it is supposed to describe the shape of query, so.. it would be not proper to have logic implementation code in here.